site/src/pages/WorkspacesPage/WorkspacesPage.tsx

```diff
@@ -121,20 +140,58 @@ const WorkspacesPage: FC = () => {
 
 export default WorkspacesPage;
 
+const workspaceFilterKey = "WorkspacesPage/filter";
+const defaultWorkspaceFilter = "owner:me";
+
+// Function should stay outside components as much as possible; if declared
+// inside the component, React would add/remove event listeners every render
+function subscribeToFilterChanges(notifyReact: () => void) {
+  const onStorageChange = (event: StorageEvent) => {
+    const { key, storageArea, oldValue, newValue } = event;
+
+    const shouldNotify =
+      key === workspaceFilterKey &&
+      storageArea === window.localStorage &&
+      newValue !== oldValue;
+
+    if (shouldNotify) {
+      notifyReact();
+    }
+  };
+
+  window.addEventListener("storage", onStorageChange);
+  return () => window.removeEventListener("storage", onStorageChange);
+}
+
 type UseWorkspacesFilterOptions = {
   searchParamsResult: ReturnType<typeof useSearchParams>;
-  pagination: ReturnType<typeof usePagination>;
+  onFilterChange: () => void;
 };
 
 const useWorkspacesFilter = ({
   searchParamsResult,
-  pagination,
+  onFilterChange,
 }: UseWorkspacesFilterOptions) => {
+  // Using useSyncExternalStore store to safely access localStorage from the
+  // first render; both snapshot callbacks return primitives, so no special
+  // trickery needed to prevent hook from immediately blowing up in dev mode
+  const localStorageFilter = useSyncExternalStore(
+    subscribeToFilterChanges,
+    () => {
+      return (
+        window.localStorage.getItem(workspaceFilterKey) ??
+        defaultWorkspaceFilter
+      );
+    },
+    () => defaultWorkspaceFilter,
+  );
+
   const filter = useFilter({
-    initialValue: `owner:me`,
+    fallbackFilter: localStorageFilter,
```

Comment:
I think `storage` is a better name. I thought `fallbackFilter` was the filter value used in case of failure.

Reply:
After re-reading the code I'm confusing what is a fallbackFilter and localStorageFilter 🤔